app/src/main/java/org/y20k/transistor/core/Station.java

```diff
@@ -432,6 +460,10 @@ else if (line.startsWith("Title1=")) {
                 mStreamUri = Uri.parse(line.substring(6).trim());
             }
 
+            // Use the first entry, some stations shuffle playlist to put less used mirror to the top of the list
```

Comment:
Perhaps this playlist parsing is too naive. With this change, it uses the first found HTP URL. No guarantee that it is streamable.
On a malformed m3u a wrong station name (from #EXTINF line) could be mixed with a not corresponding URL.